### PR TITLE
Expose Raptor cross shard organization as a table property

### PR DESCRIPTION
**Pull request by erichwang**
This also adds the property to the system table.

### CODE REVIEW

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java [hunk 1]
```diff
@@ -208,6 +211,10 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
 
         handle.getBucketCount().ifPresent(bucketCount -> properties.put(BUCKET_COUNT_PROPERTY, bucketCount));
         handle.getDistributionName().ifPresent(distributionName -> properties.put(DISTRIBUTION_NAME_PROPERTY, distributionName));
+        // Only display cross shard organization property if set
```
Comment (erichwang) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java]: How do we feel about this? Should it always be displayed?

**Comment by erichwang** [comment]
This addresses issue #6408

**Comment by electrum** [comment]
Let's call this "organization" rather than "cross shard organization"

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorOutputTableHandle.java [hunk 2]
```diff
@@ -44,6 +44,7 @@
     private final OptionalLong distributionId;
     private final OptionalInt bucketCount;
     private final List<RaptorColumnHandle> bucketColumnHandles;
+    private final boolean crossShardOrganization;
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorOutputTableHandle.java]: Name this "organizationEnabled"

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorTableProperties.java [hunk 3]
```diff
@@ -37,6 +38,7 @@
     public static final String BUCKET_COUNT_PROPERTY = "bucket_count";
     public static final String BUCKETED_ON_PROPERTY = "bucketed_on";
     public static final String DISTRIBUTION_NAME_PROPERTY = "distribution_name";
+    public static final String CROSS_SHARD_ORGANIZATION_PROPERTY = "cross_shard_organization";
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorTableProperties.java]: "organization"

File: presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/Table.java [hunk 4]
```diff
@@ -34,14 +34,22 @@
...
+    private final boolean crossShardOrganization;
 
-    public Table(long tableId, OptionalLong distributionId, Optional<String> distributionName, OptionalInt bucketCount, OptionalLong temporalColumnId)
+    public Table(
+            long tableId,
+            OptionalLong distributionId,
+            Optional<String> distributionName,
+            OptionalInt bucketCount,
+            OptionalLong temporalColumnId,
+            boolean crossShardOrganization)
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/Table.java]: "organizationEnabled"

File: presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/TableMetadataSystemTable.java [hunk 5]
```diff
@@ -186,6 +188,9 @@ public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHand
 
             // bucketing_columns
             writeArray(pageBuilder.nextBlockBuilder(), bucketColumnNames.values());
+
+            // cross_shard_organization
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/TableMetadataSystemTable.java]: Rename

File: presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/Table.java [hunk 6]
```diff
@@ -69,6 +77,11 @@ public OptionalLong getTemporalColumnId()
         return temporalColumnId;
     }
 
+    public boolean isOrganization()
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/Table.java]: This method name is a bit strange. How about `isOrganizationEnabled`?

File: presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/TableMetadataRow.java [hunk 7]
```diff
@@ -75,6 +84,11 @@ public OptionalInt getBucketCount()
         return bucketCount;
     }
 
+    public boolean isOrganization()
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/TableMetadataRow.java]: Same

File: presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/TableMetadataSystemTable.java [hunk 8]
```diff
@@ -88,7 +89,8 @@ public TableMetadataSystemTable(@ForMetadata IDBI dbi, TypeManager typeManager)
                         new ColumnMetadata("ordering_columns", arrayOfVarchar),
                         new ColumnMetadata("distribution_name", VARCHAR),
                         new ColumnMetadata("bucket_count", BIGINT),
-                        new ColumnMetadata("bucketing_columns", arrayOfVarchar)));
+                        new ColumnMetadata("bucketing_columns", arrayOfVarchar),
+                        new ColumnMetadata("organization", BOOLEAN)));
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/TableMetadataSystemTable.java]: I'm thinking `organization_enabled`

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java (hunk 1 above)
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java]: Remove "cross shard" from the comment

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorTableProperties.java [hunk 9]
```diff
@@ -63,6 +65,12 @@ public RaptorTableProperties(TypeManager typeManager)
                 .add(lowerCaseStringSessionProperty(
                         DISTRIBUTION_NAME_PROPERTY,
                         "Shared distribution name for colocated tables"))
+                .add(booleanSessionProperty(
+                        ORGANIZATION_PROPERTY,
+                        "Enable table organization of the sort order",
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorTableProperties.java]: "Keep the table organized using the sort order"

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorTableProperties.java [hunk 10]
```diff
@@ -97,6 +105,12 @@ public static String getDistributionName(Map<String, Object> tableProperties)
         return (String) tableProperties.get(DISTRIBUTION_NAME_PROPERTY);
     }
 
+    public static boolean isCrossShardOrganization(Map<String, Object> tableProperties)
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorTableProperties.java]: `isOrganizationEnabled`

**Comment by erichwang** [comment]
ok fixed up as requested. back to you for more feedback @electrum

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorTableProperties.java [hunk 11]
```diff
@@ -63,6 +65,12 @@ public RaptorTableProperties(TypeManager typeManager)
                 .add(lowerCaseStringSessionProperty(
                         DISTRIBUTION_NAME_PROPERTY,
                         "Shared distribution name for colocated tables"))
+                .add(booleanSessionProperty(
+                        ORGANIZED_PROPERTY,
+                        "Keep the table organized using the sort order",
+                        null,
+                        false
+                ))
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorTableProperties.java]: Previous line

File: presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/MetadataDao.java [hunk 12]
```diff
@@ -117,15 +117,15 @@ TableColumn getTableColumn(
             "  create_time, update_time, table_version,\n" +
             "  shard_count, row_count, compressed_size, uncompressed_size)\n" +
             "VALUES (\n" +
-            "  :schemaName, :tableName, :compactionEnabled, :organizationEnabled, :distributionId,\n" +
```
Comment (electrum) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/MetadataDao.java]: Keep the original name so the parameter matches the column name in the table

**Pull request merged**